content/blog/recipes-0-1-15/index.Rmd

```diff
@@ -0,0 +1,135 @@
+
+[Mine Çetinkaya-Rundel](https://twitter.com/minebocek) had a good idea for one such persistent issue related to creating dummy variables. For classification data where one or more predictors are categorical, the users might accidentally capture the _outcome_ and the predictors when creating dummy variables. For example: 
+
+```{r bad-inds}
+library(tidymodels)
+
+data(scat, package = "modeldata")
+
+scat_rec <- 
+  recipe(Species ~ Location + Age + Mass + Diameter, data = scat) %>% 
+  step_dummy(all_nominal(), one_hot = TRUE) %>% 
+  prep()
+
+scat_rec %>% 
+  bake(new_data = NULL) %>% 
+  names()
+```
+
+Note that the outcome column (`Species`) was made into a binary indicators. Most classification models prefer a factor vector and this would cause errors. The fix would be to remember to remove `Species` from the step selector. 
+
+Most selectors in recipes are used to capture _predictor_ columns. The new version of recipes contains new selectors that combine the role and the data type: `all_nominal_predictors()` and `all_numeric_predictors()`. Using these: 
+
+```{r good-inds}
+scat_rec <- 
+  recipe(Species ~ Location + Age + Mass + Diameter, data = scat) %>% 
+  step_dummy(all_nominal_predictors(), one_hot = TRUE) %>% 
+  prep()
+
+scat_rec %>% 
+  bake(new_data = NULL) %>% 
+  names()
+```
+
+The existing selectors will remain. We'll be converting our documentation, books, and training to use these new selectors when we select predictors of a specific type. 
+
+## New steps
+
+A new selector was added to compliment `step_rm()` (which removes columns). The new `step_select()` declares which columns to retain and emulates `dplyr::select()`. 
+
+In cases where there are missing data, some data analysis methods compliment the existing predictors with missing value indicators for the covariates that have incomplete values. Thanks to [Konrad Semsch](https://konradsemsch.netlify.com/), `step_indicate_na()` can be used to create these. Using the previous example: 
+
+```{r na-ind}
+scat_rec <- 
+  recipe(Species ~ Location + Age + Mass + Diameter, data = scat) %>% 
+  step_dummy(all_nominal_predictors(), one_hot = TRUE) %>% 
+  step_indicate_na(Mass, Diameter) %>% 
+  prep()
+
+scat_rec %>% 
+  bake(new_data = scat[!complete.cases(scat),],
+       contains("Mass"), contains("Diameter")) 
+```
+
+Speaking of missing data, we've decided to rename the current eight imputation steps:
+
+* `step_impute_knn()` is favored over `step_knnimpute()`
+* `step_impute_median()` is favored over `step_medianimpute()`
+* and so on...
+
+These are a lot better since they work well with tab-complete. The old steps will go through a gradual deprecation process before being removed at some point in the future. 
```

Comment:
```suggestion
These are a lot better since they work well with tab-completion. The old steps will go through a gradual deprecation process before being removed at some point in the future. 
```